server.js

```diff
@@ -2,22 +2,137 @@ const express = require("express");
 const cors = require("cors");
 
 const app = express();
+app.use(express.json())
 
 app.use(cors());
 
-const welcomeMessage = {
-  id: 0,
-  from: "Bart",
-  text: "Welcome to CYF chat system!",
-};
+const messages = [
+  {
+    id: 0,
+    from: "Bart",
+    text: "Welcome to CYF chat system!",
+    
+  }]
 
 //This array is our "data store".
 //We will start with one message in the array.
 //Note: messages will be lost when Glitch restarts our server.
-const messages = [welcomeMessage];
+// const messages = [welcomeMessage];
 
 app.get("/", function (request, response) {
   response.sendFile(__dirname + "/index.html");
 });
+//////////////////////Create a new message//////////////////////////// 
+ app.post('/messages/', (req, res) =>{
+  let newId = messages[messages.length -1].id + 1
+  let newMessage = {
+        id: newId,
+        from: req.body.from,
+        text: req.body.text,
+        created_at :new Date().toISOString()
+      };
+ 
+  if((newMessage.from !== undefined) && (newMessage.from.length > 0) && (newMessage.text !== undefined)&& (newMessage.from.length > 0) ){
+  messages.push(newMessage)
+  res.json(messages)
+    }
+    else{
+      res.status(400).send("Please post a validate message")
+    }
+    })
+////////////////////////Read all messages//////////////////////////
+app.get('/messages', (req, res) => {
+  res.json(messages)
+})
+/////////////////////////Read one message specified by an ID///////////////////////////
+app.get('/messages/:id', (req, res) => {
+  const mesId = Number(req.params.id);
+  const result = messages.find(ms => ms.id === mesId)
+  if (result !== undefined) {
+    res.json(result)
+  }
+  else {
+    res.status(404).send("sorry not found")
+  }
+})
+///////////////////Delete a message, by ID/////////////////////////////////////////////////
+app.delete('/messages/:id', (req, res) => {
+  const delId = Number(req.params.id);
+  const index = messages.findIndex(ms => ms.id === delId)
+  if (index !== undefined) {
+    messages.splice(index, 1)
+    res.json("DELETED!!!")
```

Comment:
I'd rather stick to a consistent messaging format. The first one for pushing a new message was very elegant. I'd rather stick to something like that.